react/features/chat/components/native/Chat.js

```diff
@@ -0,0 +1,196 @@
+// @flow
+
+import React, { Component } from 'react';
+import { Modal, SafeAreaView } from 'react-native';
+import { GiftedChat } from 'react-native-gifted-chat';
+import { connect } from 'react-redux';
+import type { Dispatch } from 'redux';
+
+import { translate } from '../../../base/i18n';
+import { getLocalParticipant } from '../../../base/participants';
+import { BackButton, Header, HeaderLabel } from '../../../base/react';
+
+import { sendMessage, toggleChat } from '../../actions';
+
+import styles from './styles';
+
+/**
+ * The type of the React {@code Component} props of {@link Chat}.
+ *
+ * Co-authored with https://github.com/dgeorgiev06
+ */
+type Props = {
+
+    /**
+     * True if the chat window should be open.
+     */
+    _isOpen: boolean,
+
+    /**
+     * The local participant.
+     */
+    _localParticipant: Object,
+
+    /**
+     * All the chat messages in the conference.
+     */
+    _messages: Array<Object>,
+
+    /**
+     * Toggles the chat window.
+     */
+    _onToggleChat: Function,
+
+    /**
+     * Send a text message.
+     *
+     * @protected
+     */
+    _onSendMessage: Function
+};
+
+
+/**
+ * The chat page of the mobile (i.e. React Native) application.
+ */
+class Chat extends Component<Props> {
+
+    /**
+       * Initializes a new Conference instance.
+       *
+       * @param {Object} props - The read-only properties with which the new
+       * instance is to be initialized.
+       */
+    constructor(props: Props) {
+        super(props);
+        this._onSend = this._onSend.bind(this);
+        this._renderMessage = this._renderMessage.bind(this);
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *x
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+
+        const messages = this.props._messages.map(this._renderMessage);
+
+        return (
+            <Modal
```

Comment:
Using a modal for a long lived dialog like this might not be the best idea. See https://developer.apple.com/design/human-interface-guidelines/ios/app-architecture/modality/

Since these will show on top of everything, no matter what, we should at least make sure it closes automatically if the conference fails.

Reply:
Good point! Also, seems like it was not handled correctly in the past either, so I added a state listener to the conference and put some actions there. Please take a minute checking that too.